osu.Game/Overlays/RankingsOverlay.cs

```diff
@@ -223,17 +220,25 @@ private void loadContent(Drawable content)
             if (content == null)
             {
                 contentContainer.Clear();
-                loading.Hide();
+                endOparation();
                 return;
             }
 
             LoadComponentAsync(content, loaded =>
             {
-                loading.Hide();
+                if (!(content is SpotlightsLayout))
```

Comment:
what's this condition for?

Reply:
So the idea is: for any display, but spotlights we starting a new operation in the overlay, because requests are handled in the overlay itself, but for spotlights will have it's own operation. When switching to spotlights, it will start an operaton in `LoadComplete` and will wait untill spotlights are loaded.
Basically we don't want to start an operation in the overlay itself because spotlights display is empty when loaded.
(I hope my explanation makes sense)

Reply:
it doesn't make sense, no. you can't start two operations at once on one operation tracker instance, it'll crash. if this conditional was reversed, it'd *maybe* make sense, but still be janky anyway due to releasing the operation and taking it back again.

Reply:
I'm not starting two operations at once though. But yeah, I see why it's too janky. I can try to expose actions from spotlights display and use them to trigger the same operation in the overlay itself

Reply:
Alright, I did so in the latest commit